### ISSUE TITLE
[BUG][C#][csharp] Use of obselete RestClient.ExecuteTaskAsync

### ISSUE THREAD

**Issue by johnthagen** [OpenAPITools/openapi-generator#11489]
#### Bug Report Checklist

- [ ] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

The `csharp` generator uses an Obsolete method:

https://github.com/OpenAPITools/openapi-generator/blob/60dcf8613f6b75b606757c4a991fe018ad10ee99/modules/openapi-generator/src/main/resources/csharp/ApiClient.mustache#L219

```cs
        [Obsolete("Use ExecuteAsync instead")]
        public virtual Task<IRestResponse> ExecuteTaskAsync(IRestRequest request, CancellationToken token) => ExecuteAsync(request, token);
```

This leads to a compiler warning:

```
  ApiClient.cs(209, 34): [CS0618] 'RestClient.ExecuteTaskAsync(IRestRequest, CancellationToken)' is obsolete: 'Use ExecuteAsync instead'
```

`RestSharp` version:

```xml
<PackageReference Include="RestSharp" Version="106.10.1" />
````

##### openapi-generator version

Current `master` / 5.4.0

##### Suggest a fix

Use `ExecuteAsync` instead.

**Comment by johnthagen** [comment]
This was fixed for `csharp-netcore` generator in #5066

**Comment by johnthagen** [comment]
`csharp-netcore` has replaced the legacy `csharp` generator as the new `csharp` name, this this issue is no longer relevant.
